### PR TITLE
#24801 adding a fix to handle content type permission inheritance

### LINKED ISSUE #24801: Workflow: Fix individual permissions
### Parent Issue
N/A

### User Story

When a limited user uses the individual permissions on a fire action, it throws an exception even if the content type has owner role with edit permission checked.

### Acceptance Criteria

- [ ] Allow limited users to publish content with a specific role
- [ ] Have to create an integration test broken to reproduce the case and once, the unit test is done, fix it

### Proposed Objective

Core Features

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://github.com/dotCMS/core/issues/24645
https://github.com/dotCMS/core/issues/24817

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by jdotcms**
The idea is when the user does not have permission over the contentlet, check if the content type has permissions

### CODE REVIEW

**Review by AndreyDotcms — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/business/WorkflowAPITest.java [hunk 1]
```diff
@@ -585,6 +586,31 @@ public static void prepare() throws Exception {
 
     }
 
+    @Test()
```
Comment (freddyucv) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/business/WorkflowAPITest.java:589]: doc

**Review by freddyucv — CHANGES_REQUESTED** [review]

**Review by freddyucv — COMMENTED** [review]
Also you have this test failing

[ExperimentAPIImpIT](https://raw.githack.com/dotCMS/test-results/issue-24801-permissions_72b0427e/projects/core/integration/postgres/reports/html/classes/com.dotcms.experiments.business.ExperimentAPIImpIT.html). [testStartExperiment_shouldPublishContent](https://raw.githack.com/dotCMS/test-results/issue-24801-permissions_72b0427e/projects/core/integration/postgres/reports/html/classes/com.dotcms.experiments.business.ExperimentAPIImpIT.html#testStartExperiment_shouldPublishContent)

it is very weird, maybe is it  a blinker?

**Comment by jdotcms** [comment]
> Also you have this test failing
> 
> [ExperimentAPIImpIT](https://raw.githack.com/dotCMS/test-results/issue-24801-permissions_72b0427e/projects/core/integration/postgres/reports/html/classes/com.dotcms.experiments.business.ExperimentAPIImpIT.html). [testStartExperiment_shouldPublishContent](https://raw.githack.com/dotCMS/test-results/issue-24801-permissions_72b0427e/projects/core/integration/postgres/reports/html/classes/com.dotcms.experiments.business.ExperimentAPIImpIT.html#testStartExperiment_shouldPublishContent)
> 
> it is very weird, maybe is it a blinker?

it is failing locally
```
Only RUNNING experiments can be ended
com.dotmarketing.business.DotStateException: Only RUNNING experiments can be ended
```

I think the test is not idempotent

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java [hunk 2]
```diff
@@ -631,8 +631,12 @@ public void save(Permission permission, Permissionable permissionable, User user
 	 */
 	@WrapInTransaction
 	private void save(Permission permission, Permissionable permissionable, User user, boolean respectFrontendRoles, boolean createEvent) throws DotDataException, DotSecurityException {
-		if(!doesUserHavePermission(permissionable, PermissionAPI.PERMISSION_EDIT_PERMISSIONS, user))
-			throw new DotSecurityException("User id: " + user.getUserId() + " does not have permission to alter permissions on asset " + permissionable.getPermissionId());
+		if(!doesUserHavePermission(permissionable, PermissionAPI.PERMISSION_EDIT_PERMISSIONS, user)) {
+
+			if(!checkIfContentletTypeHasEditPermissions(permissionable, user)) {
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java:636]: I would rename to `doesContentTypeHaveEditPermissions()`

**Review by dsilvam — APPROVED** [review]

**Pull request merged**